openmc/material.py

```diff
@@ -986,7 +986,53 @@ def get_activity(self, units: str = 'Bq/cm3', by_nuclide: bool = False):
             activity[nuclide] = inv_seconds * 1e24 * atoms_per_bcm * multiplier
 
         return activity if by_nuclide else sum(activity.values())
+    
+    def get_decayheat(self, units: str = 'W', by_nuclide: bool = False):
+        """Returns the decay heat of the material or for each nuclide in the
+        material in units of [W], [W/g] or [W/cm3].
 
+        .. versionadded:: 0.14.0
+
+        Parameters
+        ----------
+        units : {'W', 'W/g', 'W/cm3'}
+            Specifies the type of decay heat to return, options include total
+            heat [W], specific [W/g] or volumetric heat [W/cm3].
+            Default is total heat [W].
+        by_nuclide : bool
+            Specifies if the decay heat should be returned for the material as a
+            whole or per nuclide. Default is False.
+
+        Returns
+        -------
+        Union[dict, float]
+            If by_nuclide is True then a dictionary whose keys are nuclide
+            names and values are decay heat is returned. Otherwise the decay heat
+            of the material is returned as a float.
+        """
+
+        cv.check_value('units', units, {'W', 'W/g', 'W/cm3'})
+        cv.check_type('by_nuclide', by_nuclide, bool)
+
+        if units == 'W':
+            multiplier = self.volume
+        elif units == 'W/cm3':
+            multiplier = 1
+        elif units == 'W/g':
+            multiplier = 1.0 / self.get_mass_density()
+        
+        decayheat = {} 
+        for nuclide, atoms_per_bcm in self.get_nuclide_atom_densities().items():
+            decay_erg = openmc.data.decay_energy(nuclide)
+            if decay_erg is None: 
+                decayheat[nuclide] = 0.
+            else:
+                inv_seconds = openmc.data.decay_constant(nuclide)
+                decay_erg *= openmc.data.JOULE_PER_EV
+                decayheat[nuclide] = inv_seconds * decay_erg * 1e24 * atoms_per_bcm * multiplier
```

Comment:
If the default is zero, we can remove the first `if` branch
```suggestion
            inv_seconds = openmc.data.decay_constant(nuclide)
            decay_erg *= openmc.data.JOULE_PER_EV
            decayheat[nuclide] = inv_seconds * decay_erg * 1e24 * atoms_per_bcm * multiplier
```